services/lfs/server.go

```diff
@@ -495,25 +496,44 @@ func authenticate(ctx *context.Context, repository *repo_model.Repository, autho
 		accessMode = perm.AccessModeWrite
 	}
 
-	// ctx.IsSigned is unnecessary here, this will be checked in perm.CanAccess
-	perm, err := access_model.GetUserRepoPermission(ctx, repository, ctx.Doer)
-	if err != nil {
-		log.Error("Unable to GetUserRepoPermission for user %-v in repo %-v Error: %v", ctx.Doer, repository)
-		return false
-	}
+	if ctx.Data["IsActionsToken"] == true {
+		taskID := ctx.Data["ActionsTaskID"].(int64)
+		task, err := actions_model.GetTaskByID(ctx, taskID)
+		if err != nil {
+			log.Error("Unable to GetTaskByID for task[%d] Error: %v", taskID, err)
+			return false
+		}
+		if task.RepoID != repository.ID {
+			return false
+		}
 
-	canRead := perm.CanAccess(accessMode, unit.TypeCode)
-	if canRead && (!requireSigned || ctx.IsSigned) {
-		return true
-	}
+		if task.IsForkPullRequest {
+			return accessMode <= perm.AccessModeRead
+		} else {
+			return accessMode <= perm.AccessModeWrite
+		}
+	} else {
```

Comment:
Convert `ctx.Doer` into `TriggerUser`?
Not sure whether it is necessary here.

Reply:
The code in `else` block is the old code before this change. This PR is to fix the authentication for `ActionUser` so it's better to keep the old code unchanged.

Reply:
Ah, we need to convert ActionUser into TriggerUser and check the TriggerUser’s permission.
Same as #23729

Reply:
> Convert `ctx.Doer` into `TriggerUser`? 

I think it's not necessary. LFS is a part of a repo, so the workflow of a repo should be able to access its LFS objects.